scenarioColors.js

```diff
@@ -1,42 +1,62 @@
 const ScenarioSdk = require('@qasymphony/scenario-sdk');
 
-const StepSdk = {
-    getStepSdk(qtestToken, scenarioProjectId) {
-        return new ScenarioSdk.Steps({ qtestToken, scenarioProjectId });
-    }
+// @pure
```

Comment:
What's this for?

Reply:
It's just a way of rationalising the functions based on whether they are pure or impure with the idea of it making the code more easily maintainable by tracking where data is actively being mutated.  The idea comes from the following article; I'm not heavily invested in it so I'd welcome your opinion.

https://scotch.io/tutorials/wielding-pure-functions-in-javascript-and-function-composition

Reply:
I think the concept is fine, but the comments won't be maintained because developers are lazy/forgetful so it's not really worth the effort

Reply:
We've actually been pretty good at keeping our JavaDoc comments up-to-date in the test framework and check them as part of the review process.  I wouldn't see these as being any different.